### PR TITLE
Backend add auth skeleton

### PR DESCRIPTION
**Pull request by WozzyaQ**

### CODE REVIEW

File: backend/cmd/web/handlers.go [hunk 1]
```diff
@@ -1,19 +1,100 @@
...
+	if err != nil {
+		logErrorDumbExit(w, err)
+		return
+	}
+	fmt.Println(tokenStr)
+
+	w.WriteHeader(http.StatusOK)
+	resp := make(map[string]string)
+	resp["token"] = tokenStr
+	resp["message"] = "Logged in succesfully!"
```
Comment (MakTsy) [inline backend/cmd/web/handlers.go:55]: Maybe Registered, or Signed up seccesfully?

File: backend/cmd/web/users.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	"golang.org/x/crypto/bcrypt"
+)
+
+var ErrUserNotExists = errors.New("user not exists")
+
+
+type User struct {
+	ID             int
+	Name           string
+	Email          string
```
Comment (MakTsy) [inline backend/cmd/web/users.go:19]: Email is still needed?

File: backend/cmd/web/handlers.go (hunk 1 above)
Comment (WozzyaQ) [reply backend/cmd/web/handlers.go]: Fixed

File: backend/cmd/web/users.go (hunk 2 above)
Comment (WozzyaQ) [reply backend/cmd/web/users.go:19]: No, I'll remove it

File: frontend/src/redux/features/auth/authSlice.js [hunk 3]
```diff
@@ -13,12 +13,12 @@ export const registerUser = createAsyncThunk(
...
-            
-            const { data } = await axios.post('/user', {
-                username,
-                password,
+            const { data } = await axios.post('/user/signup', {
+                username, 
+                password
             })
-
+            // TODO: remove log
```
Comment (MakTsy) [inline frontend/src/redux/features/auth/authSlice.js:20]: Ready to remove the log, or next time?

**Review by MakTsy — APPROVED** [review]

File: frontend/src/redux/features/auth/authSlice.js (hunk 3 above)
Comment (WozzyaQ) [reply frontend/src/redux/features/auth/authSlice.js:20]: Let's keep it until further refactoring

**Pull request merged**